### PR TITLE
Documentation for calico kubectl plugin

### PR DESCRIPTION
**Pull request by caseydavenport**
## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: _includes/content/ctl-kubectl-plugin-install.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+you want to install the binary.
+
+   > **Note**: In order to install `{{include.cli}}` as a kubectl plugin, the binary must be located in your `PATH`. For example,
+   > `/usr/local/bin/`.
+   {: .alert .alert-info}
+
+1. Use the following command to download the `calicoctl` binary.
+
+   ```bash
+   curl -o kubectl-calico -L  {% include urls component="calicoctl" %}
```
Comment (bcreane) [inline _includes/content/ctl-kubectl-plugin-install.md:13]: this renders to `curl -o kubectl-calico -L  https://projectcalico.org/builds/calicoctl` and that path does not exist.

**Review by bcreane — COMMENTED** [review]
https://projectcalico.org/builds/calicoctl does not (currently) exist.

File: _includes/content/ctl-kubectl-plugin-install.md (hunk 1 above)
Comment (caseydavenport) [reply _includes/content/ctl-kubectl-plugin-install.md:13]: Yeah, it doesn't exist for master. That URL is a placeholder on master but resolves to the correct thing for release branches...

This is the same URL used for our binary download method, for example.

Comment (lwr20) [reply _includes/content/ctl-kubectl-plugin-install.md:13]: Sounds like we need a new post-release check for this binary.

File: _includes/content/ctl-kubectl-plugin-install.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+You can now run any `{{include.cli}}` subcommands through `kubectl calico`.
+
+> **Note**: If you run these commands from your local machine (instead of a host node), some of
+> the node related subcommands will not work (like node status).
+{: .alert .alert-info}
+
+**Next step**:
+
+[Configure `calicoctl` to connect to your datastore](configure).
```
[the remark this answers is not in the document]
Comment (caseydavenport) [reply _includes/content/ctl-kubectl-plugin-install.md:41]: That's independent of the "install as a plugin" approach - e.g,. if you want to use calicoctl as a kubectl plugin in etcd mode, you still need to configure access!

File: _includes/content/ctl-kubectl-plugin-install.md (hunk 1 above)
Comment (caseydavenport) [reply _includes/content/ctl-kubectl-plugin-install.md:13]: maybe, or we just need to change the URL so it doesn't look so "real".

e.g., `https://placeholder-for-calicoctl`

Comment (lwr20) [reply _includes/content/ctl-kubectl-plugin-install.md:13]: Well, where ever it comes from, we need a post-release check for this new release artifact :)

Comment (caseydavenport) [reply _includes/content/ctl-kubectl-plugin-install.md:13]: Oh no, it's not a new release artifact. It's the same calicoctl binary and URL. We're not introducing anything new here.

File: _includes/content/ctl-kubectl-plugin-install.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+   > kubectl to detect the plugin and allow you to use it.
+   {: .alert .alert-info}
+
+1. Verify the plugin works.
+
+   ```
+   kubectl calico -h
+   ```
+
+You can now run any `{{include.cli}}` subcommands through `kubectl calico`.
```
[the remark this answers is not in the document]
Comment (caseydavenport) [reply _includes/content/ctl-kubectl-plugin-install.md:33]: That alias sort of defeats the purpose of the kubectl plugin though. 

Running calicoctl as a kubectl plugin doesn't provide any new or changed functionality. It literally is functionally equivalent to an alias of "kubectl calico" -> "calicoctl"

[the remark this answers is not in the document]
Comment (lwr20) [reply _includes/content/ctl-kubectl-plugin-install.md:33]: OK.  Maybe just say then that anywhere in the docs it says `calicoctl`, they can now use `kubectl calico`

[the remark this answers is not in the document]
Comment (lwr20) [reply _includes/content/ctl-kubectl-plugin-install.md:33]: > It literally is functionally equivalent to an alias of "kubectl calico" -> "calicoctl"
Almost - there's also some config options its taking care of for you?  Like DATASTORE_TYPE?

[the remark this answers is not in the document]
Comment (lwr20) [reply _includes/content/ctl-kubectl-plugin-install.md:33]: You know what, ignore this.  The sentence that there now is close enough.

[the remark this answers is not in the document]
Comment (caseydavenport) [reply _includes/content/ctl-kubectl-plugin-install.md:33]: > Almost - there's also some config options its taking care of for you? Like DATASTORE_TYPE?

Nope - there is literally no functional difference. The default change for datastore type is independent of using as a kubectl plugin or not :)

**Pull request merged**